src/Symfony/Component/Process/PhpSpecifExecutableFinder.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Process;
+
+use Symfony\Component\Process\Exception\PhpSpecifExecutableInvalidVersionException;
+use Symfony\Component\Process\Exception\PhpSpecifExecutableNotFoundException;
+
+/**
+ * An executable finder specifically designed for the PHP executable.
+ * This class allows searching for a specific version of PHP if provided.
+ *
+ * @author Pululu Kinaga Andre <pululuandre@gmail.com>
+ */
+class PhpSpecifExecutableFinder
+{
+    /**
+     * Finds the PHP executable path, optionally for a specific version.
+     *
+     * @return string The path to the PHP executable.
+     * @throws PhpSpecifExecutableNotFoundException If no executable is found.
+     */
+    public function find(string $version): string
+    {
+
+        if (!$this->isValidPhpVersion($version)) {
+            throw new PhpSpecifExecutableInvalidVersionException("Invalid php version : ".$version);
+        }
+
+        $binary = "php{$version}";
+        $process = $this->runProcess(['command', '-v', $binary]);
```

Comment:
`command` won't be cross-platform. You should be using the ExecutableFinder instead (which is already implemented in a cross-platform way)